src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py

```diff
@@ -1719,3 +1720,116 @@ def skip_onnx_input_quantize(
 
     if output_file_path:
         onnx.save(model, output_file_path)
+
+def _propagate_mobilebert_embedding_quantization(model: ModelProto):
+    """
+    A pass for propagating embedding quantizations through concat
+
+    Starting with:
+    |           GATHER     (UINT8 data initializer)
+    |           |
+    |       DequantizeLinear
+    |         |   |   |
+    |         | Slice Slice
+    |         |   |   |
+    |         |  Pad Pad
+    |         |   |   |
+    |           Concat
+    |             |
+    |           OUTPUT
+
+    Converts to:
+    |           GATHER     (UINT8 data initializer)
+    |         |   |   |
+    |         | Slice Slice
+    |         |   |   |
+    |         |  Pad Pad
+    |         |   |   |
+    |           Concat
+    |             |
+    |       DequantizeLinear
+    |             |
+    |           OUTPUT
```

Comment:
note that the engine can do something like this currently for ops like reshapes and transposes, would be interesting to see if we could handle this as well

Reply:
If I understand it correctly, the issue w/ that is that the exported onnx would not convert the Linear module that follows the embeddings w/ a QMatMul.

In the ONNX export the above conversion precedes the Linear conversion to QMatMul, and the Linear conversion needs the dequantize node to be directly at the input. So this has to be done at the export level.